private void doTestInvalidProtocol(boolean throwOnFailure) throws Exception {
    Connector c = new Connector("foo.Bar");
    c.setThrowOnFailure(throwOnFailure);
    c.start();
}